bake/hclparser/hclparser.go

```diff
@@ -541,6 +547,31 @@ func (p *parser) resolveBlockNames(block *hcl.Block) ([]string, error) {
 	return names, nil
 }
 
+func (p *parser) validateVariable(v *variable, ectx *hcl.EvalContext) hcl.Diagnostics {
+	if v.Validation == nil {
+		return nil
+	}
+	condition, diags := v.Validation.Condition.Value(ectx)
+	if diags.HasErrors() {
+		return diags
+	}
+	if !condition.True() {
+		message, diags := v.Validation.Message.Value(ectx)
+		if diags.HasErrors() {
+			return diags
+		}
+		return hcl.Diagnostics{
+			&hcl.Diagnostic{
+				Severity: hcl.DiagError,
+				Summary:  "Validation failed",
+				Detail:   message.AsString(),
+				Subject:  v.Validation.Condition.Range().Ptr(),
```

Comment:
Was wondering if we should make the subject the whole var block with `v.Body.MissingItemRange().Ptr(),` or just the condition like here?